tools/autograd/gen_variable_type.py

```diff
@@ -828,6 +885,66 @@ def emit_increment_version(f: NativeFunction) -> List[str]:
             return []
         return [f'increment_version({r});' for r in cpp.return_names(f)]
 
+    def emit_fw_derivatives() -> List[str]:
+        content: List[str] = []
+        for derivative in fw_derivatives:
+            res = derivative.var_name
+
+            if_stmt = " || ".join([FW_DERIVATIVE_CHECK_TEMPLATE.substitute(req_inp=inp.name)
+                                   for inp in differentiable_inputs if inp.name in derivative.required_inputs_fw_grad])
+            if not if_stmt:
+                # Handle functions like stack
+                # For these, we don't unpack anything and always call the user function
+                assert len(differentiable_inputs) == 1
```

Comment:
This assert is very specific; it's a purposeful narrowing of functionality to ease implementation. That's fine, but there should be an English message in this case explaining exactly what assumption is being made here.